### ISSUE TITLE
Bug with template parameters expanded from a macro

### ISSUE THREAD

**Issue by zajo** [microsoft/vscode-cpptools#5783]
**Type: LanguageService**

First thank you for the excellent work, to my pleasant surprise Intellisense on Windows works nearly perfectly. I'm reporting an issue which isn't super important, but it is the only part of my library it has problems with.

For portability between C++11 and 17 I use a macro for deducing nontype template arguments:

```
template <class E, BOOST_LEAF_MATCH_ARGS(match_enum_type<E>, V1, V)>
struct is_predicate<match<E, V1, V...>>: std::true_type
{
};
```

Intellisense ignores the macro, reporting errors in that prat of the program, even though pointing at the macro with the mouse proves it is not confused about where and how the macro is defined.

For reference: the macro is defined [here](https://github.com/zajo/leaf/blob/develop/include/boost/leaf/pred.hpp?ts=4#L21) and used in several places, e.g. [here](https://github.com/zajo/leaf/blob/develop/include/boost/leaf/pred.hpp?ts=4#L107).

**Comment by sean-mcmanus** [comment]
Thanks for reporting this. I've filed a bug at https://developercommunity.visualstudio.com/content/problem/1114791/cc-intellisense-gives-error-parameter-pack-was-ref.html .

I also got some IntelliSense process crashes when I used "cl.exe /P /std:c++latest /I ...\leaf-develop\include ...\leaf-develop\test\match_test.cpp" to create a stand alone TU, so I've reported that at https://developercommunity.visualstudio.com/content/problem/1114794/cc-intellisense-process-crashes.html . I'm not sure if those crashes are something you would hit normally, but it would potentially could happen after opening certain headers that the match_test.cpp includes.

UPDATE: Fix should be in 0.30.0-insiders (no ETA yet).
